File: solutions/zig/grains/1/grains.zig
const std = @import("std");
pub const ChessboardError = error{IndexOutOfBounds};

pub fn square(index: usize) ChessboardError!u64 {
    if (index < 1 or index > 64) return ChessboardError.IndexOutOfBounds;
    return std.math.pow(u64, 2, index - 1);
}

pub fn total() ChessboardError!u64 {
    var result: u64 = 0;
    for (1..64 + 1) |idx| {
        result += square(idx) catch |err| return err;
    }
    return result;
}

File: solutions/c/grains/1/grains.c
#include "grains.h"

uint64_t square(uint8_t index){
    if (index > 0 && index <= 64) {
        return 1ULL << (index - 1);
    }
    return 0;
}
uint64_t total(void){
    return UINT64_MAX;
}